tokio/src/runtime/context.rs

```diff
@@ -1,12 +1,15 @@
 //! Thread local runtime context
+use crate::runtime::task::Id;
 use crate::runtime::{Handle, TryCurrentError};
 
-use std::cell::RefCell;
+use std::cell::{Cell, RefCell};
 
 thread_local! {
     static CONTEXT: RefCell<Option<Handle>> = RefCell::new(None)
 }
 
+thread_local! {pub(crate) static TASK_ID: Cell<Option<Id>> = Cell::new(None) }
```

Comment:
it would be preferable to use an existing thread local value to store the current task ID, if possible, because some systems have a limit on the number of thread local variables that may be created. we may want to change the existing `CONTEXT` thread-local into a `struct`, like this:

```rust
thread_local! {
    static CONTEXT: Context = Context {
        handle: RefCell::new(None),
        current_id: Cell::new(None), 
    };
}

struct Context {
    handle: RefCell<Option<Handle>>,
    current_id: Cell<Option<Id>>,
}
```
this way, we would only use a single thread-local key, rather than two.